src/api/graphql.ts

```diff
@@ -0,0 +1,19 @@
+import { createProxyMiddleware } from "http-proxy-middleware";
+import { NextApiHandler } from "next";
+import { getSession } from "next-auth/client";
+import { graphUrl } from "./config";
+
+export const graphqlProxy: NextApiHandler = async (req, res) => {
+  const session = await getSession({ req });
+  const userId = (session?.userId as string) || "";
+
+  const apiProxy: any = createProxyMiddleware({
```

Comment:
Slick! I wonder how much work goes into creating the middleware with this function. It might be worth extracting it so we don't create the middleware on every request.

Reply:
Ah it might become awkward to pass the header in but yeah agreed we should look into this.

Reply:
Very good point. Let's leave it for the future Dan and Matt

Reply:
Those 2 guys are so smart